src/rpc/connectivity/cluster.cc

```diff
@@ -363,9 +366,14 @@ void connectivity_cluster_t::run_t::connect_to_peer(
                 tls_ctx, selected_addr->ip(), selected_addr->port().value(),
                 drainer_lock.get_drain_signal(), cluster_client_port);
             if (!*successful_join_inout) {
-                handle(
+                join_result_t join_result = handle(
                     &conn, expected_id, boost::optional<peer_address_t>(*address),
-                    drainer_lock, successful_join_inout, join_delay_secs);
+                    expected_server_id, drainer_lock, successful_join_inout, join_delay_secs);
+
+                // NOTE: not an ideal way to handle this
```

Comment:
I think you could return this value from `connect_to_peer`, and then do the "merging" of the different results in a lambda function that you pass to the `pmap` in `join_blocking`.